app/assets/scripts/views/imagery-search.js

```diff
@@ -130,6 +165,25 @@ var ImagerySearch = React.createClass({
     }
   },
 
+  onMapChange: function () {
+    if (this.mapMovementThrottle) return;
+    // Using Browser History just to update the hash URL. React's hashHistory causes
+    // a scroll back to the top of the page. hashHistory is still a dependency because
+    // of the expectation that the host does not support SPA routing to index.html.
+    const path = `#/imagery-search/${this.mapViewToString()}`;
+    window.history.replaceState(null, null, path);
```

Comment:
Ah, I didn't know that. Ok, I'll change the comment to reflect that then.

BTW, am I right though about why `hashHistory` is used throughout rather than browser history API? Namely Github's lack of support for SPA routing.